libsolidity/ast/AST.h

```diff
@@ -496,7 +499,13 @@ class ContractDefinition: public Declaration, public StructurallyDocumented, pub
 	std::vector<EnumDefinition const*> definedEnums() const { return filteredNodes<EnumDefinition>(m_subNodes); }
 	std::vector<VariableDeclaration const*> stateVariables() const { return filteredNodes<VariableDeclaration>(m_subNodes); }
 	std::vector<ModifierDefinition const*> functionModifiers() const { return filteredNodes<ModifierDefinition>(m_subNodes); }
-	std::vector<FunctionDefinition const*> definedFunctions() const { return filteredNodes<FunctionDefinition>(m_subNodes); }
+	auto definedFunctions() const {
+		return definedFunctionsByName() | ranges::views::transform([](auto const& _item) { return _item.second; });
```

Comment:
```suggestion
		return definedFunctionsByName() | ranges::views:: keys;
```
does that work?

Reply:
Yes, that should work, when including ``<range/v3/view/map.hpp>`` and is nicer.

Reply:
I tried it, but it doesn't - it's a multimap - d'oh!

Reply:
No really, I'm underwhelmed by the power of that library.

Reply:
It does work for me for a multimap :-).

Reply:
At least this:
```
#include <range/v3/all.hpp>
#include <iostream>
#include <map>

int main()
{

        std::multimap<int, int> x{{1,1},{2,3},{2,4}};
        auto r = x | ranges::views::values;
        for (auto v : r)
                std::cout << v << std::endl;
        std::cout << std::endl;

        auto&& [b,e] = x.equal_range(2);
        auto r2 = ranges::subrange(b,e) | ranges::views::values;
        for (auto v : r2)
                std::cout << v << std::endl;
}
```
compiles just fine for me.

Reply:
Weird...